File: backend/app/schemas/document.py
import uuid
from datetime import datetime
from decimal import Decimal

from pydantic import BaseModel


class DocumentResponse(BaseModel):
    id: uuid.UUID
    tenant_id: uuid.UUID
    location_id: uuid.UUID | None
    source: str
    original_filename: str
    mime_type: str
    file_size_bytes: int
    status: str
    document_type: str
    document_date: datetime | None
    vendor_name: str | None
    total_amount: Decimal | None
    currency_code: str
    is_duplicate: bool
    duplicate_of: uuid.UUID | None
    created_by: uuid.UUID
    created_at: datetime
    updated_at: datetime
    download_url: str | None = None

    model_config = {"from_attributes": True}


class OCRResultResponse(BaseModel):
    id: uuid.UUID
    provider: str
    extracted_text: str
    confidence_score: Decimal
    page_count: int
    processing_time_ms: int | None
    processed_at: datetime

    model_config = {"from_attributes": True}


class LineItemResponse(BaseModel):
    id: uuid.UUID
    description: str
    quantity: Decimal | None
    unit_price: Decimal | None
    amount: Decimal
    currency_code: str
    confidence_score: Decimal
    manually_corrected: bool

    model_config = {"from_attributes": True}


class LineItemCorrectionRequest(BaseModel):
    description: str | None = None
    amount: Decimal | None = None
    quantity: Decimal | None = None
    unit_price: Decimal | None = None

File: frontend/src/types/document.ts
export type DocumentStatus =
  | "pending"
  | "ocr_processing"
  | "ocr_complete"
  | "extracting"
  | "extracted"
  | "categorized"
  | "reconciled"
  | "error";

export type DocumentType = "invoice" | "receipt" | "bill" | "statement" | "other";

export interface Document {
  id: string;
  tenant_id: string;
  location_id: string | null;
  source: string;
  original_filename: string;
  mime_type: string;
  file_size_bytes: number;
  status: DocumentStatus;
  document_type: DocumentType;
  document_date: string | null;
  vendor_name: string | null;
  total_amount: string | null;
  currency_code: string;
  is_duplicate: boolean;
  duplicate_of: string | null;
  created_by: string;
  created_at: string;
  updated_at: string;
  download_url?: string | null;
}

export interface LineItem {
  id: string;
  description: string;
  quantity: string | null;
  unit_price: string | null;
  amount: string;
  currency_code: string;
  confidence_score: string;
  manually_corrected: boolean;
}

export interface OCRResult {
  id: string;
  provider: string;
  extracted_text: string;
  confidence_score: string;
  page_count: number;
  processing_time_ms: number | null;
  processed_at: string;
}
